docs/modules/SDDs/pages/0022-managed-services-in-cluster.adoc

```diff
@@ -0,0 +1,111 @@
+* In the case of having a Kubernetes Cluster which only provides an application service, a central GitOps repository can be used which stores all application instance objects. As the user of the application service doesn't get access to the cluster, there is no way of tampering with the Argo CD configuration.
+** An alternative could be to create a GitOps repository per customer which contains all the service instance objects and allow the customer to do some kind of self-service.
+* A separate Argo CD project will be created for deploying the application services, this way there is a grouping and clear boundary available.
+
+=== Risks and Mitigations
+
+==== Operator Updates and Upgrades
+
+There is a risk when upgrading a Kubernetes Operator that the service instance could get restarted, redeployed, deleted, destroyed, or anything else you could imagine as the Kubernetes Operator is the responsible tool taking care of the application service.
+
+To mitigate this a defined test procedure is needed which makes sure that Kubernetes Operator upgrades are safe. This also includes figuring out if objects have changed and are still compatible.
+
+It could also be that object conversion are needed befor a Kubernetes Operator can be upgraded which would need additional tooling.
+
+==== Version mismatch
+
+As the service instance object is managed in a different way and outside of the reach of the Commodore Component of the Kubernetes Operator, it could happen that the instance object requests a version which is not supported by the Kubernetes Operator. Or that after an upgrade the version isn't support anymore.
+
+It is therefore important to keep track of the service instances and their versions and keep them in sync and comparison to the Kubernetes Operator version. This is something which might be implemented done via the Lieutenant API and inventory.
+
+== Drawbacks
+
+=== Service instantiation
+
+With this approach Project Syn currently only handles the Kubernetes Operator itself including some plumbing, but it doesn't handle the actual service instantiation.
+
+== Alternatives
+
+=== Crossplane
+
+One of the cornerstone of Project Syn will be Crossplane to provide fully integrated service provisioning. At the current state Crossplane will mainly be used for provisioning services in the cloud. For provisioning of services directly in the cluster there is still more work to do. Also Crossplane is currently not production ready yet and would need a lot of work to get it integrated. For this reason directly using an application service specific Operator it is considered good enough until Crossplane has matured enough. The end-goal will be to use Crossplane for all-things service provisioning, be it in the cloud or in the cluster.
```

Comment:
I do not consider Crossplane to be an alternative. As far as the concept discussed within this SDD, Crossplane is just an operator and all of the above does apply.

Reply:
Yeah, right. I'll replace that by other alternatives.